internal/task/task.go

```diff
@@ -0,0 +1,333 @@
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("{}"))
+	}
+}
+
+// Обработчик удаления задачи по ID
+func DeleteTaskHandler(db *sql.DB) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		id := r.URL.Query().Get("id")
+		if id == "" {
+			sendErrorResponse(w, "Не указан идентификатор задачи")
+			return
+		}
+
+		query := `DELETE FROM scheduler WHERE id = ?`
+		res, err := db.Exec(query, id)
+		if err != nil {
+			sendErrorResponse(w, fmt.Sprintf("Ошибка при удалении задачи: %v", err))
+			return
+		}
+
+		rowsAffected, err := res.RowsAffected()
+		if err != nil {
+			sendErrorResponse(w, "Ошибка при получении информации об удалении")
+			return
+		}
+
+		if rowsAffected == 0 {
+			sendErrorResponse(w, "Задача не найдена")
+			return
+		}
+
+		// Успешный ответ
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("{}"))
+	}
+}
+
+// Обработчик завершения задачи
+func PostTaskDoneHandler(db *sql.DB) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		id := r.URL.Query().Get("id")
+		if id == "" {
+			sendErrorResponse(w, "Не указан идентификатор задачи")
+			return
+		}
+
+		var task Task
+		err := db.QueryRow(`SELECT id, date, repeat FROM scheduler WHERE id = ?`, id).
+			Scan(&task.ID, &task.Date, &task.Repeat)
+		if err == sql.ErrNoRows {
+			sendErrorResponse(w, "Задача не найдена")
+			return
+		} else if err != nil {
+			sendErrorResponse(w, fmt.Sprintf("Ошибка при запросе задачи: %v", err))
+			return
+		}
+
+		// Если повторений нет, удаляем задачу
+		if task.Repeat == "" {
+			_, err := db.Exec(`DELETE FROM scheduler WHERE id = ?`, id)
+			if err != nil {
+				sendErrorResponse(w, fmt.Sprintf("Ошибка при удалении задачи: %v", err))
+				return
+			}
+		} else {
+			// Рассчитываем следующую дату выполнения
+			now := time.Now()
+			nextDate, err := nextdate.NextDate(now, task.Date, task.Repeat)
+			if err != nil {
+				sendErrorResponse(w, fmt.Sprintf("Ошибка в правиле повторения: %v", err))
+				return
+			}
+
+			// Обновляем дату в базе
+			_, err = db.Exec(`UPDATE scheduler SET date = ? WHERE id = ?`, nextDate, id)
+			if err != nil {
+				sendErrorResponse(w, fmt.Sprintf("Ошибка при обновлении даты: %v", err))
+				return
+			}
+		}
+
+		// Успешный ответ
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("{}"))
+	}
+}
+
+// Обработчик удаления задачи по ID
+func DeleteTaskDoneHandler(db *sql.DB) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		id := r.URL.Query().Get("id")
+		if id == "" {
+			sendErrorResponse(w, "Не указан идентификатор задачи")
+			return
+		}
+
+		query := `DELETE FROM scheduler WHERE id = ?`
+		res, err := db.Exec(query, id)
+		if err != nil {
+			sendErrorResponse(w, fmt.Sprintf("Ошибка при удалении задачи: %v", err))
```

Comment:
Можно было создать еще одну функцию для ответа с ошибкой внутри добавлять ошибку в конец
```suggestion
			sendErrorResponseErr(w, "Ошибка при удалении задачи", err)
```

Reply:
@ivannizh Спасибо большое за ревью. Изменения внес) Учту опыт на будущее. Можно еще попросить комментарий по поводу структуры документа? Что можно было бы улучшить/изменить?

Наверно через годик взгляну на проект и буду плакать )

Reply:
Почитай про реализацию Чистой архитектуры в Го. У тебя можно БД вынести в отдельный слой, а так же бизнес логику